crypto/http/http_lib.c

```diff
@@ -112,14 +138,36 @@ int OSSL_HTTP_parse_url(const char *url, char **phost, char **pport,
         ERR_raise(ERR_LIB_HTTP, HTTP_R_INVALID_URL_PATH);
         goto parse_err;
     }
+    path_end = query = query_end = frag = frag_end = path + strlen(path);
+
+    /* parse optional "?query" */
+    tmp = strchr(path, '?');
+    if (pquery != NULL && tmp != NULL) {
+        path_end = tmp;
+        query = path_end + 1;
+        query_end = frag = frag_end = query + strlen(query);
```

Comment:
Why do you think that `query + strlen(query)` will give a different result than `path + strlen(path)` on line 141?  It looks to me like this line is superfluous...

Reply:
Good point - I had put the line for symmetry with the above, not realizing that it is redundant.
Removed, as well as the one below.